src/tests/Microsoft.PowerFx.Interpreter.Tests.Shared/RecalcEngineTests.cs

```diff
@@ -679,33 +681,56 @@ public void ImperativeUserDefinedFunctionTest(string udfExpression, string expre
         [Fact]
 
         public void DelegatableUDFTest()
-        {
-            var config = new PowerFxConfig();
-            config.EnableSetFunction();
-
+        {
+            var symbolTable = new DelegatableSymbolTable();
             var schema = DType.CreateTable(
                 new TypedName(DType.Guid, new DName("ID")),
                 new TypedName(DType.Number, new DName("Value")));
-            config.SymbolTable.AddEntity(new TestDataSource("MyDataSource", schema));
-            config.SymbolTable.AddType(new DName("MyDataSourceTableType"), FormulaType.Build(schema));
+            symbolTable.AddEntity(new TestDelegableDataSource(
+                "MyDataSource",
+                schema,
+                new TestDelegationMetadata(
+                        new DelegationCapability(DelegationCapability.Filter | DelegationCapability.Count | DelegationCapability.Sort),
+                        schema,
+                        new FilterOpMetadata(
+                            schema,
+                            new Dictionary<DPath, DelegationCapability>(),
+                            new Dictionary<DPath, DelegationCapability>(),
+                            new DelegationCapability(DelegationCapability.Equal | DelegationCapability.GreaterThanOrEqual),

```

Comment:
>DelegationCapability.Equal | DelegationCapability.GreaterThanOrEqual [](http://example.com/codeflow?start=53&length=68)

The delegation capabilities for this data source is for the `=` and the `>=`  operators, but in the expressions below you are using `>`. Should the UDF be marked as delegatable? If not, why is the test passing - is it a bug?

Reply:
Yes, it appears to be a bug on this check in `IsSupportedOpNode`.

`        public virtual bool IsSupportedOpNode(TexlNode node, OperationCapabilityMetadata metadata, TexlBinding binding)
        {
            Contracts.AssertValue(node);
            Contracts.AssertValue(metadata);
            Contracts.AssertValue(binding);

            var binaryOpNode = node.AsBinaryOp();
            if (binaryOpNode == null)
            {
                return false;
            }

            var opDelStrategy = _function.GetOpDelegationStrategy(binaryOpNode.Op, binaryOpNode);
            var binaryOpDelStrategy = (opDelStrategy as BinaryOpDelegationStrategy).VerifyValue();
            Contracts.Assert(binaryOpNode.Op == binaryOpDelStrategy.Op);

            // Check if binaryOp is supported by datasource in the context of filter operation.
            // If this is not allowed then there is no point in checking lhs and rhs
            // It's only safe to do so if lhs and rhs is first/dotted name node as columns (FirstName/DottedName node) can have additional capabilities defined.
            if (!(binaryOpNode.Left is FirstNameNode || binaryOpNode.Left is DottedNameNode) &&
                !(binaryOpNode.Right is FirstNameNode || binaryOpNode.Right is DottedNameNode) &&
                !opDelStrategy.IsOpSupportedByTable(metadata, node, binding))`

Reply:
for binaryOp `Value > 10` it shortcircuits at `!(binaryOpNode.Left is FirstNameNode || binaryOpNode.Left is DottedNameNode)`, and I am not sure what the point of that check is

Reply:
Will file bug